### PR TITLE
[R] various improvements and bug fixes

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh`, `./bin/security/{LANG}-petstore.sh` and `./bin/openapi3/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

- use optional parameters
- fix header parameters
- rename callAPI to CallAPI to conform to the style guide
- fix API key authentication

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/r/api.mustache [hunk 1]
```diff
@@ -93,13 +93,13 @@
       {{#isApiKey}}
```
Comment (Ramanth) [inline modules/openapi-generator/src/main/resources/r/api.mustache]: in the lines above at line 91, the usage of jsonlite::base64_enc() is not working when the authentication string is longer.
The issue is that the above method  is encoding but adding a extra '\n' in the output of the encoded string.

Please refer to a similar issue in the below link:
[jsonline issue](https://github.com/jeroen/jsonlite/issues/220)

Please make a fix or any workaround. Thanks in advance

Comment (wing328) [reply modules/openapi-generator/src/main/resources/r/api.mustache]: Thanks. Will use other base64 encoding function instead. Can you please open an issue to track this instead?

Comment (wing328) [reply modules/openapi-generator/src/main/resources/r/api.mustache]: @Ramanth I've filed https://github.com/OpenAPITools/openapi-generator/pull/2171 to fix the issue. Can you please check to see if that works for you?

Comment (wing328) [reply modules/openapi-generator/src/main/resources/r/api.mustache]: #2171 merged into r_refactor. Please give it a try.

**Pull request merged**